### PR TITLE
Toxic sent

### PR DESCRIPTION
**Pull request by aceroyuno**
## About The Pull Request
This pull request is to add the new Sentinel strain Toxic Sentinel. At it's core it slashes enemies to deal toxic damage at the cost of plasma, it uses utility abilities to aid in this endeavor.
Further elaborated on in here https://docs.google.com/document/d/1Ez3sAHZ3RhoPE3K-Dnuq6pQTk3U8p_Rxm8qj7TqEeHs/edit?usp=sharing.

## Why It's Good For The Game
There are a lot of reasons that this is good for the game, adds a new T1 strain for a caste that does not have any.
As of right now, most marines carry ina, bica, and kelo auto injectors. This provides an indirect counter to that metaplay.
It allows xenos to have another type of damage source.

## Changelog

:cl: Aceroy
qol: Xenos can now purchase the "Toxic" strain as a Sentinel
Add: Added new strain for Sentinel "Toxic"
Add: Added Sentinel blinding spit ability (Toxic Sentinel) - A ranged spit attack that causes a little over a second of blindness. It has a range of 5 tiles. Costs 40 plasma.
Add: Added Sentinel sprint ability (Toxic Sentinel) - A movement speed stim that lasts 30 seconds and costs 40 plasma.
Add: Added Sentinel toxic slash togglable ability (Toxic Sentinel) - A togglable passive that causes each slash to deal 10 toxic damage per slash. It costs 30 plasma to deal the 10 toxic damage.
Balance: Sentinel gains 60 max health when becoming Toxic Sentinel
/:cl:

### CODE REVIEW

**Comment by stanalbatross** [comment]
More detailed changelog will be needed

**Comment by stanalbatross** [comment]
Also pr your Google doc as an actual design doc to the design doc repo

File: code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm [hunk 1]
```diff
@@ -106,3 +106,94 @@
...
+/datum/action/xeno_action/onclick/toggle_toxic_slash/use_ability(atom/target)
+	var/mob/living/carbon/Xenomorph/xeno = owner
+	if(!istype(xeno))
+		return
+	var/datum/behavior_delegate/sentinel_toxic/behavior = xeno.behavior_delegate
+	if(!istype(behavior))
+		return
+	if(behavior.toxic_toggle)
+		behavior.toxic_toggle = FALSE
+		to_chat(xeno, SPAN_XENOHIGHDANGER("Your slashes will now inject cytotoxin into the victim"))
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm:121]:
```suggestion
		to_chat(xeno, SPAN_XENOHIGHDANGER("Your slashes will now inject cytotoxin into the victim."))
```

File: code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm [hunk 2]
```diff
@@ -106,3 +106,94 @@
...
+		return
+	if(!behavior.toxic_toggle)
+		behavior.toxic_toggle = TRUE
+		to_chat(xeno, SPAN_XENOHIGHDANGER("Your slashes will no longer inject cytotoxin into the victim"))
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm:125]:
```suggestion
		to_chat(xeno, SPAN_XENOHIGHDANGER("Your slashes will no longer inject cytotoxin into the victim."))
```

File: code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm [hunk 3]
```diff
@@ -106,3 +106,94 @@
...
+	projectile.generate_bullet(xeno.ammo)
+	projectile.permutated += xeno
+	projectile.def_zone = xeno.get_limbzone_target()
+	projectile.fire_at(target, xeno, xeno, xeno.ammo.max_range, xeno.ammo.shell_speed)
+
+	apply_cooldown()
+	..()
+
+/datum/action/xeno_action/onclick/sentinel_sprint/use_ability(atom/A)
+	var/mob/living/carbon/Xenomorph/X = owner
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm:163]: Var name

File: code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm [hunk 4]
```diff
@@ -106,3 +106,94 @@
...
+	X.recalculate_speed()
+
+	addtimer(CALLBACK(src, .proc/remove_effects), duration)
+
+	apply_cooldown()
+	..()
+	return
+
+/datum/action/xeno_action/onclick/sentinel_sprint/proc/remove_effects()
+	var/mob/living/carbon/Xenomorph/X = owner
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm:190]: Var name

File: code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm [hunk 5]
```diff
@@ -106,3 +106,94 @@
...
+	to_chat(X, SPAN_XENOHIGHDANGER("You feel yourself move quicker!"))
+	buffs_active = TRUE
+	X.speed_modifier -= speed_buff_amount
+	X.recalculate_speed()
+
+	addtimer(CALLBACK(src, .proc/remove_effects), duration)
+
+	apply_cooldown()
+	..()
+	return
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm:187]:
```suggestion
	return ..()
```

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/sentinel/toxic.dm [hunk 6]
```diff
@@ -0,0 +1,52 @@
+/datum/xeno_mutator/toxic
+	name = "STRAIN: Sentinel - Toxic"
+	description = "all of your nuerotoxin is replaced with cytotoxins"
+	flavor_description = "Ssssss I am a snek"
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/mutators/strains/sentinel/toxic.dm:4]: This is shown to players so needs to be better

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/sentinel/toxic.dm (hunk 6 above)
Comment (aceroyuno) [reply code/modules/mob/living/carbon/xenomorph/mutators/strains/sentinel/toxic.dm]: Updated description and flavor description.

File: code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm (hunk 5 above)
Comment (aceroyuno) [reply code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm]: Fixed

File: code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm (hunk 4 above)
Comment (aceroyuno) [reply code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm]: Changed var name to be inline with the rest of the file.

File: code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm (hunk 3 above)
Comment (aceroyuno) [reply code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm]: changed var name to be in line with the rest of the file.

File: code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm (hunk 2 above)
Comment (aceroyuno) [reply code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm]: Fixed punctuation error.

File: code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm (hunk 1 above)
Comment (aceroyuno) [reply code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm]: fixed punctuation error

**Comment by GoldenDarkness55** [comment]
The blindness is cool, but strain's probably useless though - see acid runner's performance. Slugs are too powerful to have a working melee t1. Nno stuns or slows whatsoever doesnt help much either, remember even t1s are supposed to win 1v1s without much issue.
Should probably fix vomit stun duration/thresholds for that and tox organ damage aswell cause this could get silly.
Also isn't this just going to be tedious and annoying? Current healing kits got introduced so you dont have to run to a medic for every single boo-boo - but with this you'll have to ask a medic everytime you get slashed by a sent? Theres a reason toxic dmg was discussed plenty but never implemented

File: code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm [hunk 7]
```diff
@@ -106,3 +106,93 @@
...
+	var/mob/living/carbon/Xenomorph/xeno = owner
+
+	if (!action_cooldown_check())
+		return
+
+	if (!istype(xeno) || !xeno.check_state())
+		return
+
+	if (buffs_active)
+		to_chat(xeno, SPAN_XENOHIGHDANGER("You cannot stack frenzy!"))
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/abilities/sentinel/sentinel_powers.dm:172]:
```suggestion
		to_chat(xeno, SPAN_XENOHIGHDANGER("You cannot stack Sprint!"))
```

**Pull request closed without merging**